src/main/java/seedu/address/logic/commands/DeleteCommand.java

```diff
@@ -4,43 +4,43 @@
 
 import java.util.List;
 
-import seedu.address.commons.core.index.Index;
 import seedu.address.commons.util.ToStringBuilder;
 import seedu.address.logic.Messages;
 import seedu.address.logic.commands.exceptions.CommandException;
 import seedu.address.model.Model;
+import seedu.address.model.person.IdentityCardNumberMatchesPredicate;
 import seedu.address.model.person.Person;
 
 /**
- * Deletes a person identified using it's displayed index from the address book.
+ * Deletes a person identified using it's displayed identity card number from the address book.
  */
 public class DeleteCommand extends Command {
 
     public static final String COMMAND_WORD = "delete";
 
     public static final String MESSAGE_USAGE = COMMAND_WORD
-            + ": Deletes the person identified by the index number used in the displayed person list.\n"
-            + "Parameters: INDEX (must be a positive integer)\n"
-            + "Example: " + COMMAND_WORD + " 1";
+            + ": Deletes the person identified by the identity card number used in the displayed person list.\n"
+            + "Parameters: IC (must be a valid identity card number)\n"
+            + "Example: " + COMMAND_WORD + " S1234567A";
 
     public static final String MESSAGE_DELETE_PERSON_SUCCESS = "Deleted Person: %1$s";
 
-    private final Index targetIndex;
+    private final IdentityCardNumberMatchesPredicate predicate;
 
-    public DeleteCommand(Index targetIndex) {
-        this.targetIndex = targetIndex;
+    public DeleteCommand(IdentityCardNumberMatchesPredicate predicate) {
+        this.predicate = predicate;
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
         List<Person> lastShownList = model.getFilteredPersonList();
 
-        if (targetIndex.getZeroBased() >= lastShownList.size()) {
-            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
-        }
+        Person personToDelete = lastShownList.stream()
+                .filter(predicate::test)
+                .findFirst()
+                .orElseThrow(() -> new CommandException(Messages.MESSAGE_NO_MATCHING_IC));
```

Comment:
@jiahui0309 does this mean that you'll only be able to delete patients that are currently visible on the screen?

Reply:
Currently, it uses a list of profiles which are previously filtered, but might be better to use the entire database of patients in the list

Reply:
Agreed, thanks!